DQM/Physics/src/TopSingleLeptonDQM_miniAOD.cc

```diff
@@ -494,25 +504,38 @@ void MonitorEnsemble::fill(const edm::Event& event,
              max(0., neHadEt + phoEt - 0.5 * muon->pfIsolationR04().sumPUPt)) /
             muon->pt();  // CB dBeta corrected iso!
 
+        if(!(muon->isGlobalMuon() && muon->isPFMuon() && muon->globalTrack()->normalizedChi2() < 10. && muon->globalTrack()->hitPattern().numberOfValidMuonHits() > 0 && muon->numberOfMatchedStations() > 1 && fabs(muon->muonBestTrack()->dxy(PV.position())) < 0.2  && fabs(muon->muonBestTrack()->dz(PV.position())) < 0.5 && muon->innerTrack()->hitPattern().numberOfValidPixelHits() > 0 && muon->innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5   ) )continue;   
+
+        //tight id
         if (mMult == 0) {
           // restrict to leading muon
-          fill("muonPt_", muon->pt());
-          fill("muonEta_", muon->eta());
-
           fill("muonRelIso_", pfRelIso);
 
           fill("muonChHadIso_", chHadPt);
           fill("muonNeHadIso_", neHadEt);
           fill("muonPhIso_", phoEt);
         }
         ++mMult;
-        if (!muonIso_ || (*muonIso_)(*muon)) ++mMultIso;
+        //if (!muonIso_ || (*muonIso_)(*muon)) ++mMultIso;
+        if(pfRelIso<0.15)++mMultIso;
+
+        if(!(pfRelIso<0.15 &&  muon->isGlobalMuon() && muon->isPFMuon() && muon->globalTrack()->normalizedChi2() < 10. && muon->globalTrack()->hitPattern().numberOfValidMuonHits() > 0 && muon->numberOfMatchedStations() > 1 && fabs(muon->muonBestTrack()->dxy(PV.position())) < 0.2  && fabs(muon->muonBestTrack()->dz(PV.position())) < 0.5 && muon->innerTrack()->hitPattern().numberOfValidPixelHits() > 0 && muon->innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5   ) )continue;
```

Comment:
most of this if statement is repeated just above - which means most criteria are satisfied by definition - those should all be dropped.

second - move the if ( mTight==0) above to avoid unneeded calculations.